apps/prairielearn/src/pages/instructorStudents/instructorStudents.html.tsx

```diff
@@ -0,0 +1,226 @@
+  parseAsColumnPinningState,
+  parseAsColumnVisibilityStateWithColumns,
+  parseAsSortingState,
+} from '../../lib/client/nuqs.js';
+import type { StaffCourseInstanceContext } from '../../lib/client/page-context.js';
+
+import { ColumnManager } from './components/ColumnManager.js';
+import { DownloadButton } from './components/DownloadButton.js';
+import { StudentsTable } from './components/StudentsTable.js';
+import { type StudentRow } from './instructorStudents.shared.js';
+
+// This default must be declared outside the component to ensure referential
+// stability across renders, as `[] !== []` in JavaScript.
+const DEFAULT_SORT: SortingState = [];
+
+const columnHelper = createColumnHelper<StudentRow>();
+
+interface StudentsCardProps {
+  students: StudentRow[];
+  timezone: string;
+  courseInstance: StaffCourseInstanceContext['course_instance'];
+  course: StaffCourseInstanceContext['course'];
+}
+
+function StudentsCard({ students, timezone, courseInstance, course }: StudentsCardProps) {
+  const [globalFilter, setGlobalFilter] = useQueryState('search', parseAsString.withDefault(''));
+  const [sorting, setSorting] = useQueryState<SortingState>(
+    'sort',
+    parseAsSortingState.withDefault(DEFAULT_SORT),
+  );
+  const [columnPinning, setColumnPinning] = useQueryState(
+    'frozen',
+    parseAsColumnPinningState.withDefault({ left: ['user_uid'], right: [] }),
+  );
+
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  // Focus the search input when Ctrl+F is pressed
+  useEffect(() => {
+    function onKeyDown(event: KeyboardEvent) {
+      if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'f') {
+        if (searchInputRef.current && searchInputRef.current !== document.activeElement) {
+          searchInputRef.current.focus();
+          event.preventDefault();
+        }
+      }
+    }
+    document.addEventListener('keydown', onKeyDown);
+    return () => document.removeEventListener('keydown', onKeyDown);
+  }, []);
+
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const [columnSizing, setColumnSizing] = useState<ColumnSizingState>({});
+
+  const columns = useMemo<AccessorKeyColumnDef<StudentRow, any>[]>(
+    () => [
+      columnHelper.accessor('user.uid', {
+        header: 'UID',
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor('user.name', {
+        header: 'Name',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('user.email', {
+        header: 'Email',
+        cell: (info) => info.getValue() || '—',
+      }),
+      columnHelper.accessor('enrollment.created_at', {
+        header: `Enrolled on (${formatTz(timezone)})`,
+        cell: (info) => {
+          const date = new Date(info.getValue());
+          return (
+            <div style={{ fontVariantNumeric: 'tabular-nums' }}>
+              {formatDate(date, timezone, {
+                includeTz: false,
+              })}
```

Comment:
In new code I think we should standardize on user-visible dates being formatted with `formatDateFriendly()` from https://github.com/PrairieLearnInc/PrairieTest/blob/41e3f354a10114db1226e701f690e4f98ded8dd1/src/lib/formatter.ts#L508

We should still use ISO 8601 dates for import/export, of course.

Reply:
This is a little interesting as it would mean that `@prairielearn/formatter` depends on `react-bootstrap`. Is that ok?

Reply:
IMO no, that would feel super weird to me and I'd very much rather avoid that. If this is a pattern we want, we can implement it locally in `components/` or something like that.

Reply:
Should I do that?

Reply:
The utils were pulled out into #12299 and then I will make a tooltip component wrapping that.